### PR TITLE
Rate limit counter metrics points to one per second

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
High throughput metrics were recently rate limited (https://github.com/solana-labs/solana/pull/5447) by default to avoid dropping them during high TPS. This fixed one issue but created a new problem. Now in low TPS environments, if the rate limit is set too high and metrics are overly aggregated, the dashboard reports crazy numbers: https://cdn.discordapp.com/attachments/439194979856809985/610149325300957201/unknown.png (reported by @mvines)

#### Summary of Changes
Updated the counter metrics rate limiting mechanism to stop limiting by the number of times a counter was incremented and start limiting by time passed.

- Counter metrics are now bucketed into 1s data points by default (env var overridable)

#### Notes
- This new approach is more adaptive and should work well in high and low TPS scenarios
- The metrics writer max datapoints rate is 4000/s so counters shouldn't push this over the limit anymore

### CODE REVIEW

File: metrics/src/counter.rs [hunk 1]
```diff
@@ -230,36 +164,29 @@ impl Counter {
...
-                    submit(point.to_owned(), level);
-                }
-            }
+        let lastlog = self.lastlog.load(Ordering::Relaxed);
+        let prev = self
+            .lastlog
+            .compare_and_swap(lastlog, counts, Ordering::Relaxed);
+        if prev == lastlog {
+            let bucket = now / metricsrate;
+            let counter = CounterPoint {
```
Comment (sakridge) [inline metrics/src/counter.rs]: I think we found allocating like this is kind of slow, hence we are using the cached point, is that possible here?

Comment (sakridge) [reply metrics/src/counter.rs]: I guess maybe not since we are sending to a channel.

Comment (jstarry) [reply metrics/src/counter.rs]: Oh interesting, are there differences in allocation performance between cloning an existing struct and allocating a new one?

Comment (jstarry) [reply metrics/src/counter.rs]: Anyways, we are allocating less with this change. `CounterPoint` is much lighter weight than `influxdb::Point`

File: metrics/src/counter.rs [hunk 2]
```diff
@@ -313,31 +240,31 @@ mod tests {
 
     #[test]
     #[serial]
-    fn test_high_rate_counter() {
+    fn test_metrisrate() {
```
Comment (mvines) [inline metrics/src/counter.rs]: "metris" rate?

Comment (jstarry) [reply metrics/src/counter.rs]: Yeah, it's that new Tetris spinoff everyone's been raving about!

**Pull request merged**